### PR TITLE
add realtime loader for metnotes

### PR DESCRIPTION
**Pull request by RousseauLambertLP**
add realtime loader for metnotes

see msc-geomet issue 466

### CODE REVIEW

File: msc_pygeoapi/loader/__init__.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def data():
     ('msc_pygeoapi.loader.aqhi_realtime', 'aqhi_realtime'),
     ('msc_pygeoapi.loader.ltce', 'ltce'),
     ('msc_pygeoapi.loader.climate_archive', 'climate_archive'),
+    ('msc_pygeoapi.loader.metnotes', 'metnotes'),
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/__init__.py:59]: remove trailing spaces from end of dicts/sets/tuples

File: msc_pygeoapi/loader/metnotes.py [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+from pathlib import Path
+
+import click
+from elasticsearch import logger as elastic_logger
+
+from msc_pygeoapi import cli_options
+from msc_pygeoapi.connector.elasticsearch_ import ElasticsearchConnector
+from msc_pygeoapi.env import (
+    MSC_PYGEOAPI_CACHEDIR,
+    MSC_PYGEOAPI_LOGGING_LOGLEVEL,
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:44]: remove trailing spaces from end of dicts/sets/tuples

File: msc_pygeoapi/loader/metnotes.py [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+                    },
+                    'metnote_status': {
+                        'type': 'text',
+                        'fields': {
+                            'raw': {
+                                'type': 'keyword'
+                            }
+                        }
+                    },
+                    'filename': {
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:116]: What does an example `filename` look like?

File: msc_pygeoapi/loader/metnotes.py [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+
+
+        self.update_temporal_config()
+        # update metnote status to active
+        self.set_active_metnote()
+
+        return True
+
+    def update_es_index(self, es_index, id, feature):
+        '''
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:228]: `s/'''/"""/g`

File: msc_pygeoapi/loader/metnotes.py [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+                refresh=True
+            )
+            LOGGER.debug('Result: {}'.format(r))
+            return True
+        except Exception as err:
+            LOGGER.warning('Error indexing: {}'.format(err))
+            return False
+
+    def update_temporal_config(self):
+        '''
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:249]: `s/'''/"""/g`

File: msc_pygeoapi/loader/metnotes.py [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+        Create/update the temporal configuration file
+        for the metnotes
+        returns: `bool` of update status
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:252]: Put empty lines between descriptoins, params and returns

```
def foo(var1):
    """
    My foo function

    :param var1: some string

    :returns: some transformed string
    """
```

File: msc_pygeoapi/loader/metnotes.py [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+        '''
+        updated_config = None
+
+        latest_time = self.filename.split('_')[0]
+        latest_file_time = datetime.strptime(latest_time, 
+            DATETIME_FORMAT)
+
+        if self.config_file.exists():
+            with open(self.config_file, 'r') as f:
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:261]: Do we need `r` given it's the default mode?

File: msc_pygeoapi/loader/metnotes.py [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+                                'latest_file': self.filename,
+                                'latest_file_time': latest_time
+                             }
+            self.latest_file = self.filename
+
+        if not updated_config:
+            return False
+
+        LOGGER.info(
+            f'Updating metnotes temporal configuration file...'
```
Comment (tomkralidis) [inline msc_pygeoapi/loader/metnotes.py:292]: Does not need an f-string

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: msc_pygeoapi/loader/metnotes.py (hunk 3 above)
Comment (RousseauLambertLP) [reply msc_pygeoapi/loader/metnotes.py:116]: sure, filename is the filename for the metnote file on HPFX, eg:

**Review by tomkralidis — APPROVED** [review]

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**